src/main/java/com/getindata/connectors/http/internal/table/lookup/LookupQueryInfo.java

```diff
@@ -0,0 +1,60 @@
+package com.getindata.connectors.http.internal.table.lookup;
+
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import lombok.Getter;
+import lombok.ToString;
+
+import com.getindata.connectors.http.internal.utils.uri.NameValuePair;
+import com.getindata.connectors.http.internal.utils.uri.URLEncodedUtils;
+
+/**
+ * Holds the lookup query for an HTTP request.
+ * The {@code  lookupQuery} either contain the query parameters for a GET operation
+ * or the payload of a body-based request.
+ * The {@code bodyBasedUrlQueryParams} contains the optional query parameters of a
+ * body-based request in addition to its payload supplied with {@code  lookupQuery}.
+ */
+@ToString
+public class LookupQueryInfo implements Serializable {
+    @Getter
+    private final String lookupQuery;
+
+    private final Map<String, String> bodyBasedUrlQueryParams;
+
+    public LookupQueryInfo(String lookupQuery) {
+        this(lookupQuery, null);
+    }
+
+    public LookupQueryInfo(String lookupQuery, Map<String, String> bodyBasedUrlQueryParams) {
+        this.lookupQuery =
+                Optional.ofNullable(lookupQuery)
+                        .orElse("");
+        this.bodyBasedUrlQueryParams =
+                Optional.ofNullable(bodyBasedUrlQueryParams)
+                        .orElse(Collections.emptyMap());
+    }
```

Comment:
Actually I don't think optional are the best thing to use here.
We are not gaining much from them and I would argue they are used not as Optionala were intend to be used.

I see that you are using them to have "nicely looking" null checks, but I think below code would do the same thing and will be easier to read than optionals.

````
this.lookupQuery = lookupQuery == null ? "" : lookupQuery;
this.bodyBasedUrlQueryParams = bodyBasedUrlQueryParams == null ? Collections.emptyMap() : bodyBasedUrlQueryParams;
````

Please substitute Optionals with the above code.

Reply:
No problem, change done in https://github.com/getindata/flink-http-connector/pull/79/commits/d0e1822704c0bc9a48df7c1611cdfcfd0e6cbb5b#diff-5d0d1d1012fdd57bd2aa4b4fe50dca8d1ce442236c2eea7283222092d917b4a4R34-R38